pkg/auth/handler/webapp/change_password.go

```diff
@@ -80,6 +118,39 @@ func (h *ChangePasswordHandler) ServeHTTP(w http.ResponseWriter, r *http.Request
 	})
 
 	ctrl.PostAction("", func() error {
+		if maybeWebSession != nil {
```

Comment:
Seems we cannot use the existence of web session to determine this is from setting page / force changing password during authentication. For some cases in the setting page there is web session too. To reproduce the problem:
case 1. Submit wrong password in the change password setting page, web session will be started.
case 2. Do other actions before entering the change password setting page, for example, create totp.

Is it correct if we change to check the request path instead?